README.md

```diff
@@ -143,19 +145,32 @@ NodeJS `Buffer` is also acceptable because it is a subclass of `Uint8Array`.
 
 #### `DecoderOptions`
 
-Name|Type|Default
-----|----|----
-extensionCodec | ExtensionCodec | `ExtensionCodec.defaultCodec`
-context | user-defined | -
-useBigInt64 | boolean | false
-maxStrLength | number | `4_294_967_295` (UINT32_MAX)
-maxBinLength | number | `4_294_967_295` (UINT32_MAX)
-maxArrayLength | number | `4_294_967_295` (UINT32_MAX)
-maxMapLength | number | `4_294_967_295` (UINT32_MAX)
-maxExtLength | number | `4_294_967_295` (UINT32_MAX)
+| Name           | Type           | Default                                                                              |
+| -------------- | -------------- | ------------------------------------------------------------------------------------ |
+| extensionCodec | ExtensionCodec | `ExtensionCodec.defaultCodec`                                                        |
+| context        | user-defined   | -                                                                                    |
+| useBigInt64    | boolean        | false                                                                                |
+| intMode        | IntMode        | `IntMode.AS_ENCODED` if `useBigInt64` is `true` or `IntMode.UNSAFE_NUMBER` otherwise |
+| maxStrLength   | number         | `4_294_967_295` (UINT32_MAX)                                                         |
+| maxBinLength   | number         | `4_294_967_295` (UINT32_MAX)                                                         |
+| maxArrayLength | number         | `4_294_967_295` (UINT32_MAX)                                                         |
+| maxMapLength   | number         | `4_294_967_295` (UINT32_MAX)                                                         |
+| maxExtLength   | number         | `4_294_967_295` (UINT32_MAX)                                                         |
 
 You can use `max${Type}Length` to limit the length of each type decoded.
 
+`intMode` determines whether decoded integers should be returned as numbers or bigints in different circumstances. The possible values are [described below](#intmode).
+
+##### `IntMode`
+
+The `IntMode` enum defines different options for decoding integers. They are described below:
+
+- `IntMode.UNSAFE_NUMBER`: Always returns the value as a number. Be aware that there will be a loss of precision if the value is outside the range of `Number.MIN_SAFE_INTEGER` to `Number.MAX_SAFE_INTEGER`.
+- `IntMode.SAFE_NUMBER`: Always returns the value as a number, but throws an error if the value is outside of the range of `Number.MIN_SAFE_INTEGER` to `Number.MAX_SAFE_INTEGER`.
+- `IntMode.MIXED`: Returns all values inside the range of `Number.MIN_SAFE_INTEGER` to `Number.MAX_SAFE_INTEGER` as numbers and all values outside that range as bigints.
+- `IntMode.AS_ENCODED`: Returns all values encoded as int64/uint64 as bigints and all other integers as numbers.
```

Comment:
I added this new int mode, which matched the BigInt support that was added to the main library.

Reply:
(and felt like a useful mode too)

Reply:
Seems useful, good idea